YAFCui/ImGui/ImGuiUtils.cs

```diff
@@ -337,7 +337,7 @@ public struct InlineGridBuilder : IDisposable {
         public static bool BuildSearchBox(this ImGui gui, SearchQuery searchQuery, out SearchQuery newQuery, string placeholder = "Search") {
             newQuery = searchQuery;
             if (gui.BuildTextInput(searchQuery.query, out var newText, placeholder, Icon.Search)) {
-                newQuery.SetSearch(newText);
+                newQuery = new SearchQuery(newText);
```

Comment:
This one is https://github.com/have-fun-was-taken/yafc-ce/pull/64/commits/559847e35741e11a105a5f864654bb9eb9693125
>  Manual refactor: make SearchQuery readonly

> Due to how SetSearch was implemented SearchQuery could not be made
readonly. However, Visual Studio would occasionally bug out and
make it readonly when running Code Cleanup, so that raised a choice:
whether to mute the inspection or to refactor the class.

> I refactored the class by removing SetSearch and making SearchQuery
readonly.

Reply:
Yes I saw, I reviewed the separate commits. I was just wondering about the impact when the user is doing some searches.

Reply:
The previous implementation essentially did the [same](https://github.com/have-fun-was-taken/yafc-ce/commit/559847e35741e11a105a5f864654bb9eb9693125#diff-aab5f4c0184c7f6e8dcafd6e14852be6ba530f125c9fb464c146501dcb7ebcaaL10) under another function, so there should be no visible changes to the performance.

Reply:
In the previous implementation the `SearchQuery` object was reused (for some reason). With the proposed one the `SearchQuery` is thrown away and a new one created (each time you type a new char in the search box?)

Reply:
The previous implementation had the following line:
```
// YAFCui/Core/SearchQuery.cs
public void SetSearch(string query) {
    this = new SearchQuery(query);
}
```
Based on this line, I assume that the `SearchQuery` object was not reused because `this` was assigned to a new object. Is my assumption incorrect?

Reply:
hm... I am not used to a language allowing to change `this`,`self`, etc... (it is kind of weird if you think about it)
But I guess it indeed creates a new object at the least (discarding the existing one)